controllers/datadogagent/feature/dogstatsd/feature.go

```diff
@@ -239,6 +243,15 @@ func (f *dogstatsdFeature) ManageNodeAgent(managers feature.PodTemplateManagers)
 		if f.udsEnabled {
 			managers.PodTemplateSpec().Spec.HostPID = true
 		}
+		// Tag cardinality is only configured if origin detection is enabled.
+		// Only three values (`low`, `orchestrator` and `high`) can be configured, others are ignored.
+		switch v2alpha1.TagCardinalityName(f.tagCardinality) {
+		case v2alpha1.LowTagCardinality, v2alpha1.OrchestratorTagCardinality, v2alpha1.HighTagCardinality:
```

Comment:
since we already use a defined type for `TagCardinalityName`, i don't think switch/case is needed anymore and could just do `if f.tagCardinality != nil`

Reply:
Due to the niche feature and the fact there is already an Agent validation, we defer to the latter instead of adding an additional validation in the Agent. This simplifies the code by removing the alias type and the switch case. Addressed in https://github.com/DataDog/datadog-operator/pull/936/commits/d5e41d371eba46b837bd531beb48d064cb78eaca